src/InstrumentationEngine.Attach/InstrumentationConfigurationSources.xsd

```diff
@@ -0,0 +1,60 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
+    <!-- root type -->
+    <xs:element name="InstrumentationConfigurationSources">
+        <xs:complexType>
+            <xs:sequence>
+                <xs:element name="InstrumentationConfigurationSource" maxOccurs="unbounded" type="instrumentationConfigurationSourceType" />
+            </xs:sequence>
+        </xs:complexType>
+    </xs:element>
+
+    <!-- level 1 types -->
+    <xs:complexType name="instrumentationConfigurationSourceType">
+        <xs:sequence>
+            <xs:element name="Platforms" type="platformsType" />
+            <xs:element name="Settings" minOccurs="0" type="settingsType" />
+        </xs:sequence>
+    </xs:complexType>
+
+    <!-- level 2 types -->
+    <xs:complexType name="platformsType">
+        <xs:sequence>
+            <xs:element name="Platform" minOccurs="1" maxOccurs="unbounded" type="platformType">
+            </xs:element>
+        </xs:sequence>
+    </xs:complexType>
+
+    <xs:complexType name="settingsType">
+        <xs:sequence>
+            <xs:element name="Setting" maxOccurs="unbounded" type="settingType" >
+            </xs:element>
+        </xs:sequence>
+    </xs:complexType>
+
+    <!-- level 3 types -->
+    <xs:complexType name="platformType">
+        <xs:attribute name="Chip" use="required" type="chipType" />
+        <xs:attribute name="Path" use="required" type="pathType" />
+    </xs:complexType>
+
+    <xs:complexType name="settingType">
+        <xs:attribute name="Name" type="xs:string" use="required" />
+        <xs:attribute name="Value" type="xs:string" use="required" />
+    </xs:complexType>
+
+    <!-- level 4 types -->
+    <xs:simpleType name="chipType">
+        <xs:restriction base="xs:string">
+            <xs:pattern value="x86|x64" />
+        </xs:restriction>
+    </xs:simpleType>
```

Comment:
I would make this:

<xs:simpleType name="ChipType">
    <xs:restriction base="xs:string">
        <xs:enumeration value="x86"/>
        <xs:enumeration value="x64"/>
    </xs:restriction>
  </xs:simpleType>

This will generate a C# enum in the backing code file.

Reply:
Thanks for the tip!

---
In reply to: [388444652](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r388444652) [](ancestors = 388444652)